### PR TITLE
[native] Update CI image and Advance Velox

### PR DESCRIPTION
**Pull request by majetideepak**
## Description

Update CI image to include CMake changes from Velox.
Advance Velox.

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by xiaoxmeng — APPROVED** [review]
@majetideepak thank you!

**Comment by majetideepak** [comment]
@wanglinsong Can you please upload `docker.io/majetideepak4/prestocpp:root-20240527` to the Presto AWS account?
The CI is passing with that image. This image was built using `podman compose build centos-native-dependency` after advancing Velox.
Thanks!

**Comment by wanglinsong** [comment]
This is published as public.ecr.aws/oss-presto/prestocpp:root-20240527.

> @wanglinsong Can you please upload `docker.io/majetideepak4/prestocpp:root-20240527` to the Presto AWS account? The CI is passing with that image. This image was built using `podman compose build centos-native-dependency` after advancing Velox. Thanks!

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ executors:
   build:
     docker:
       # These images are managed by the Presto Release team.
-      - image: public.ecr.aws/oss-presto/prestocpp:root-20240416
+      - image: docker.io/majetideepak4/prestocpp:root-20240527
```
Comment (wanglinsong) [inline .circleci/continue_config.yml:64]: After https://github.com/prestodb/presto/pull/22806 is merged, you can use the new dependency image - `public.ecr.aws/oss-presto/presto-native-dependency:latest`

See: https://us-east-1.console.aws.amazon.com/ecr/repositories/public/932483864676/presto-native-dependency?region=us-east-1

**Comment by majetideepak** [comment]
@amitkdutta, @xiaoxmeng can you approve this PR? Thanks.

**Review by xiaoxmeng — APPROVED** [review]

**Comment by majetideepak** [comment]
@tdcmeehan can you please approve this PR?

**Review by amitkdutta — APPROVED** [review]
Looks good.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**